Mathlib/Topology/Instances/ENNReal.lean

```diff
@@ -1433,6 +1433,21 @@ lemma limsup_toReal_eq {ι : Type*} {F : Filter ι} [NeBot F] {b : ℝ≥0∞} (
   rw [key]
   rfl
 
+/-- If `filter.limsup u at_top ≤ x`, then for all `ε > 0`, eventually we have `u a < x + ε`.  -/
+theorem eventually_lt_add_pos_of_limsup_le {α : Type _} [Preorder α] {x : ℝ} {u : α → ℝ}
+    (hu_bdd : IsBoundedUnder LE.le atTop u) (hu : Filter.limsup u atTop ≤ x) {ε : ℝ} (hε : 0 < ε) :
+    ∀ᶠ a : α in atTop, u a < x + ε :=
```

Comment:
```suggestion
/-- If `Filter.limsup u at_top ≤ x`, then for all `ε > 0`, eventually we have `u a < x + ε`.  -/
theorem eventually_lt_add_pos_of_limsup_le {α : Type _} [Preorder α] {x : ℝ} {u : α → ℝ}
    (hu_bdd : IsBoundedUnder LE.le atTop u) (hu : Filter.limsup u atTop ≤ x) {ε : ℝ} (hε : 0 < ε) :
    ∀ᶠ a : α in atTop, u a < x + ε :=
```
1. we would also want the `liminf` version.
2. probably we should have this as an `iff` and then make an `alias`?
3. this is about `ℝ`-valued functions, why is it in `ENNReal`?

Reply:
1. Done.
2. I don't see how to make it into an `iff` without adding the extra hypothesis `(hu_bdd : IsCoboundedUnder LE.le atTop u)`.
3. `Topology.Instances.Real` does not have all the required imports; this `ENNReal` file was probably suggested by `find_home`. I have moved this lemma and the following one to `Topology.Algebra.Order.LiminfLimsup` and replaced `ℝ` by a `ConditionallyCompleteLinearOrder`.